Gemfile

```diff
@@ -41,6 +44,8 @@ group :development do
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
+  gem 'shoulda-matchers'
+  gem 'rails-controller-testing'
```

Comment:
I'm not sure, but I don't think you need to add this to both `:development, :test` and `test`. I think you could just add them to the `:test` one. Might need to check that though.